Shopping List/ShoppingItemController.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  ShoppingItemController.swift
+//  Shopping List
+//
+//  Created by Nathanael Youngren on 1/18/19.
+//  Copyright © 2019 Lambda School. All rights reserved.
+//
+
+import UIKit
+
+class ShoppingItemController {
+    private(set) var shoppingList: [ShoppingItem] = []
+    
+    init() {
+        createItems()
+        loadFromPersistence()
+    }
+    
+    func createItems() {
+        let names = ["apple", "grapes", "milk", "muffin", "popcorn", "soda", "strawberries"]
+        for name in names {
+            let newItem = ShoppingItem(image: name, name: name.capitalized)
+            shoppingList.append(newItem)
+        }
+    }
+    
+    func updateIsAdded(item: ShoppingItem) {
+        guard let index = shoppingList.index(of: item) else { return }
+        shoppingList[index].isAdded = !shoppingList[index].isAdded
+        saveToPersistence()
+    }
+    
+    var addedItems: [ShoppingItem] {
+        let addedItems = shoppingList.filter({ $0.isAdded == true })
+        return addedItems
+    }
+    
+    var notAddedItems: [ShoppingItem] {
+        let notAddedItems = shoppingList.filter({ $0.isAdded == false })
+        return notAddedItems
+    }
+    
+    
+    func saveToPersistence() {
+        guard let url = shoppingListURL else { return }
+
+        let encoder = PropertyListEncoder()
+
+        do {
+            let listData = try encoder.encode(shoppingList)
+            try listData.write(to: url)
+        } catch {
+            print(error)
+        }
+    }
+
+    func loadFromPersistence() {
+        guard let url = shoppingListURL else { return }
```

Comment:
Before running code also checking existence of the url path is a good practice
```suggestion
        guard let url = shoppingListURL, FileManager.default.fileExists(atPath: url.path) else { return }
```